src/app/shared/components/shared.module.ts

```diff
@@ -0,0 +1,12 @@
+import { CommonModule } from '@angular/common';
+import { NgModule } from '@angular/core';
+import { FooterComponent } from './footer/footer.component';
+import { HeaderComponent } from './header/header.component';
+import { CourseCardComponent } from './course-card/course-card.component';
+
+@NgModule({
+  imports: [CommonModule],
+  declarations: [FooterComponent, HeaderComponent, CourseCardComponent],
+  exports: [FooterComponent, HeaderComponent, CourseCardComponent],
+})
+export class SharedModule {}
```

Comment:
please remove this module and create a separate module for each shared entity instead.
Your structure should look similar to this one.
Also, please remove `types` folder. Create `models` instead and use the following naming convention.
Put `course-page.models.ts` file in the same folder with `course-page.component.ts`. And create all relative interfaces, types to course-page in that file.
In case you need some global models (which I don't recommend doing), you can create the `models` folder in the `app`. Like on the screen/ But better to keep everything incapsulated as much as possible. 
![image](https://user-images.githubusercontent.com/26470573/97848967-885b0400-1cfa-11eb-899e-335e7def852b.png)

Reply:
The core folder is unnecessary right now. You will place some core functionality in that folder which you don't have yet.

Reply:
done